### ISSUE TITLE
Parameter count different from reported count in publication

### ISSUE THREAD

**Issue by kildarj** [Zhishe-Wang/MLP-Net#2]
Hi Dr Wang,

Thanks for sharing your code.

When running your training script (train.py) and outputting a parameter count using the torchinfo.summary method, the trainable parameters number 28.4 million, as opposed to the 8 million reported in your paper. Are you able to please explain this discrepancy?

Thanks in advance,
Jordan

**Comment by wangxiaofa0082** [comment]
<img width="1467" height="770" alt="Image" src="https://github.com/user-attachments/assets/0cdc8375-928e-42d9-992b-cb93c1adebe1" />

MLPnet.py
net = MLPNet(.to(device)
input_shape =(1,3,256,256) 
print("params: %s" %(params)) 
8 million

**Comment by kildarj** [comment]
Thanks for the clarification there. The real issue was in the calculation of parameters using the sum(p.numel() for p in net.parameters()) approach, as opposed to the thop profile approach (counts only forward pass implementation). The majority of modules initialised in the MLPNet class are not implemented, leading to a bloated count. There are 28 million additional parameters initialised in the model from unused modules using the default arguments. After removing these, the parameter counts are much more consistent. I am sure that I missed some other unused modules that are causing the leftover difference. 

However, the default parameters in the __init__ method for MLPNet are not the same as those specified in the train,.py. These are also different from those present in the test.py file. 

E.g. in train.py:
`if model_type == 'MLPNet':
     logger.info('*.*加载训练开始*.*')
    # #    model = UMLP(size=config.img_size, n_channels=config.n_channels, n_classes=config.n_labels, in_channels=64)
     model = MLPNet(img_size=config.img_size, patch_size=4, in_chans=config.n_channels, num_classes=config.n_labels,
                   embed_dim=64, layers=[2, 2, 2, 2], drop_rate=0.5,
                  norm_layer=nn.BatchNorm2d, alpha=3, use_dropout=False, patch_norm=True)
`
where config.img_size=256, config.n_channels=3, and config.n_labels=1000.
Compared to the default arguments:
`def __init__(self, img_size=256, patch_size=4, in_chans=3, num_classes=1000,
                 embed_dim=64, layers=[4, 4, 4, 4], drop_rate=0.5,
                 norm_layer=nn.BatchNorm2d, alpha=3, use_dropout=False, patch_norm=True, **kwargs):`
Similarly, in the test.py file:
`model = MLPNet(img_size=config.img_size, patch_size=4, in_chans=config.n_channels, num_classes=config.n_labels,
                   embed_dim=64, layers=[1, 1, 1, 1], drop_rate=0.5,
                   norm_layer=nn.BatchNorm2d, alpha=3, use_dropout=False, patch_norm=True)`

The difference is in the `layers` argument, which defaults to 4 per block, as opposed to 2 in the train configuration, and 1 in the test configuration. Can you shed some light on why those are different, and the effect this has on the model performance?
